rclcpp/include/rclcpp/intra_process_manager.hpp

```diff
@@ -368,6 +370,22 @@ class IntraProcessManager
     }
   }
 
+  /// Return true if the given rmw_gid_t matches any stored Publishers.
+  bool
+  matches_any_publishers(const rmw_gid_t * id)
+  {
+    for (auto & publisher_pair : publishers_) {
+      auto publisher = publisher_pair.second.publisher.lock();
+      if (!publisher) {
+        continue;
+      }
+      if (*publisher.get() == id) {
```

Comment:
Nvm. Just saw that below the publisher implements specific equal operators for that...